src/chrome/client/chromeDebugAdapter/connectingCDA.ts

```diff
@@ -25,8 +28,18 @@ export class ConnectingCDA extends BaseCDAState {
     }
 
     public async connect(telemetryPropertyCollector: ITelemetryPropertyCollector): Promise<ConnectedCDA> {
-        const result = await this._debuggeeLauncher.launch(this._configuration.args, telemetryPropertyCollector);
-        await this._chromeConnection.attach(result.address, result.port, result.url, this._configuration.args.timeout, this._configuration.args.extraCRDPChannelPort);
+
+        if (this._configuration.scenarioType === ScenarioType.Launch) {
+            logger.verbose('[ConnectingCDA]: Launching debugee...');
+            const result = await this._debuggeeLauncher.launch(this._configuration.args, telemetryPropertyCollector);
+            await this._chromeConnection.attach(result.address, result.port, result.url, this._configuration.args.timeout, this._configuration.args.extraCRDPChannelPort);
+        }
+        else if (this._configuration.scenarioType === ScenarioType.Attach) {
+            logger.verbose('[ConnectingCDA]: Attaching to an existing instance of debugee...');
+            const attachArgs = <IAttachRequestArgs>this._configuration.args;
+            await this._chromeConnection.attach(attachArgs.address, attachArgs.port, attachArgs.url, attachArgs.timeout, attachArgs.extraCRDPChannelPort);
+        }
```

Comment:
I'd add an else case and throw an exception or something if it's neither Launch or Attach (which should be impossible, yet I've seen impossible things happening in software before...)